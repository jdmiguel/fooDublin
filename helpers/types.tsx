import { ReactNode } from 'react';

export type InitialAppState = {
  favorites: Restaurant[];
  relatedRestaurants: Restaurant[];
  breadcrumbs: BreadcrumbsData[];
};

// Core / ui components

export enum LogoSize {
  BIG = 'big',
  SMALL = 'small',
}

export enum LoaderType {
  CIRCLE = 'circle',
  LINE = 'line',
}

export enum LoaderMode {
  DARK = 'dark',
  LIGHT = 'light',
}

export type ListItem = {
  iconSrc?: string;
  id: number;
  name: string;
  path: string;
};

export enum CardType {
  SUGGESTION = 'suggestion',
  STANDART = 'standart',
  HIGHLIGHT = 'highlight',
}

export enum CustomLinkSize {
  BIG = 'big',
  SMALL = 'small',
}

// Restaurant data

export type Timming = {
  id: string;
  day: string;
  schedule: string;
};

export enum EntityType {
  CITY = 'city',
  SUBZONE = 'subzone',
}

export type RestaurantsRequestParams = {
  entity_id: number | null;
  cuisines: number | null;
  entity_type: EntityType.CITY | EntityType.SUBZONE;
  start?: number;
  sort?: string;
  order?: string;
  q?: string;
  count?: number;
};

export type Restaurant = {
  id: number;
  imgSrc: string;
  title: string;
  content: ReactNode | string;
  route: string;
  asRoute: string;
};

export type RawRestaurant = {
  restaurant: {
    R: {
      has_menu_status: {
        delivery: number;
        takeaway: number;
      };
      res_id: number;
      is_grocery_store: boolean;
    };
    apikey: string;
    id: number;
    name: string;
    url: string;
    location: {
      address: string;
      locality: string;
      city: string;
      city_id: number;
      latitude: number;
      longitude: number;
      zipcode: string;
      country_id: number;
      locality_verbose: string;
    };
    switch_to_order_menu: number;
    cuisines: string;
    timings: string;
    average_cost_for_two: number;
    price_range: number;
    currency: string;
    highlights: string[];
    offers: any[];
    opentable_support: number;
    is_zomato_book_res: number;
    mezzo_provider: string;
    is_book_form_web_view: number;
    book_form_web_view_url: string;
    book_again_url: string;
    thumb: string;
    user_rating: {
      aggregate_rating: number;
      rating_text: string;
      rating_color: string;
      rating_obj: {
        title: {
          text: number;
        };
        bg_color: {
          type: string;
          tint: number;
        };
      };
      votes: number;
    };
    all_reviews_count: number;
    photos_url: string;
    photo_count: number;
    menu_url: string;
    featured_image: string;
    has_online_delivery: number;
    is_delivering_now: number;
    store_type: string;
    include_bogo_offers: boolean;
    deeplink: string;
    is_table_reservation_supported: number;
    has_table_booking: number;
    events_url: string;
    phone_numbers: string;
    all_reviews: {
      reviews: [
        {
          review: any[];
        },
        {
          review: any[];
        },
        {
          review: any[];
        },
        {
          review: any[];
        },
        {
          review: any[];
        },
      ];
    };
    establishment: string[];
  };
};

export type RestaurantDetail = {
  imgSrc: string;
  thumbSrc: string;
  name: string;
  location: string;
  cuisines: string;
  timings: string;
  rating: number;
  votes: number;
  average: string;
  establishment: string;
  highlights: string[];
  phone: string;
  address: string;
};

export type Review = {
  id: number;
  userName: string;
  userImgSrc: string;
  rating: number;
  date: string;
  text: string;
};

export type RawReview = {
  review: {
    rating: number;
    review_text: string;
    id: number;
    rating_color: string;
    review_time_friendly: string;
    rating_text: string;
    timestamp: number;
    likes: number;
    user: {
      name: string;
      foodie_level: string;
      foodie_level_num: number;
      foodie_color: string;
      profile_url: string;
      profile_image: string;
      profile_deeplink: string;
    };
    comments_count: number;
  };
};

// Breadcrumbs

export enum BreadcrumbsType {
  HOME = 'home',
  SEARCH = 'search',
  DETAIL = 'detail',
  FAVORITES = 'favorite',
}

export type BreadcrumbsData = {
  text: string;
  route: string;
  asRoute: string;
  type:
    | BreadcrumbsType.HOME
    | BreadcrumbsType.SEARCH
    | BreadcrumbsType.DETAIL
    | BreadcrumbsType.FAVORITES;
};

// Actions

type AddFavoriteAction = {
  type: 'ADD_FAVORITE';
  favorite: Restaurant;
};

type DeleteFavoriteAction = {
  type: 'DELETE_FAVORITE';
  id: number;
};

type SetRelatedRestaurantsAction = {
  type: 'SET_RELATED_RESTAURANTS';
  relatedRestaurants: Restaurant[];
};

type ClearRelatedRestaurantsAction = {
  type: 'CLEAR_RELATED_RESTAURANTS';
};

type AddBreadcrumbsAction = {
  type: 'ADD_BREADCRUMBS';
  breadcrumbs: BreadcrumbsData;
};

type ReplaceBreadcrumbsAction = {
  type: 'REPLACE_BREADCRUMBS';
  index: number;
  breadcrumbs: BreadcrumbsData;
};

export type Actions =
  | AddFavoriteAction
  | DeleteFavoriteAction
  | SetRelatedRestaurantsAction
  | ClearRelatedRestaurantsAction
  | AddBreadcrumbsAction
  | ReplaceBreadcrumbsAction;
