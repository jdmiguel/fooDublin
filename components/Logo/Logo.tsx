import React from 'react';
import styled from 'styled-components';

type LogoProps = {
  className?: string;
  logoSrc: string;
  logoAltTxt: string;
};

const StyledImg = styled.img`
  max-width: 300px;
`;

const Logo: React.FC<LogoProps> = ({ className, logoSrc, logoAltTxt }) => (
  <h1 className={className} data-testid="logo">
    <StyledImg src={logoSrc} alt={logoAltTxt} />
  </h1>
);

export default Logo;
