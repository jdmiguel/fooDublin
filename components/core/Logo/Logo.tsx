import React from 'react';
import Link from 'next/link';
import styled from 'styled-components';

type LogoProps = {
  className?: string;
  logoSrc: string;
  isLink?: boolean;
};

const StyledLink = styled.a`
  margin-bottom: 5px;
  @media only screen and (min-width: 540px) {
    margin-bottom: 7px;
  }
`;

const StyledImg = styled.img`
  max-width: 300px;
`;

const getLogoContent = (className: string | undefined, logoSrc: string) => (
  <h1 className={className} data-testid="logo">
    <StyledImg src={logoSrc} alt="FooDublin Logo" />
  </h1>
);

const Logo: React.FC<LogoProps> = ({ className, logoSrc, isLink = false }) => (
  <>
    {isLink ? (
      <Link href="/" passHref>
        <StyledLink>{getLogoContent(className, logoSrc)}</StyledLink>
      </Link>
    ) : (
      getLogoContent(className, logoSrc)
    )}
  </>
);

export default Logo;
