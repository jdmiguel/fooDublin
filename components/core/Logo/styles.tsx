import styled from 'styled-components';
import { LogoSize } from '../types';

export const StyledLogo = styled.h1<{ size: LogoSize.BIG | LogoSize.SMALL }>`
  width: ${({ size }) => (size === LogoSize.BIG ? '225px' : '190px')};
  margin: ${({ size }) => size === LogoSize.BIG && '15px 0 25px'};
  @media only screen and (min-width: 768px) {
    width: ${({ size }) => size === LogoSize.BIG && '270px'};
  }
  @media only screen and (min-width: 992px) {
    width: ${({ size }) => (size === LogoSize.BIG ? '300px' : '200px')};
    margin: ${({ size }) => size === LogoSize.BIG && '0 0 30px'};
  }
`;

export const StyledImg = styled.img`
  max-width: 300px;
`;
