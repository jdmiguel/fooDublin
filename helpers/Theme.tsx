import React from 'react';
import { DefaultTheme, ThemeProvider } from 'styled-components';

export const theme: DefaultTheme = {
  palette: {
    PRIMARY: '#EC7E00',
    PRIMARY_DARK: '#D48C24',
    PRIMARY_MEDIUM: '#FBA52B',
    PRIMARY_LIGHT: '#FDF3E5',
    SECONDARY: '#660000',
    LIGHT_MAX: '#FFFFFF',
    LIGHT_MEDIUM: '#F3F3F3',
    LIGHT_MIN: '#E5EAED',
    LIGHT_SOFT: '#D8D8D8',
    DARK_MAX: '#222C3C',
    DARK_MEDIUM: '#585858',
    DARK_MIN: '#89959B',
    DARK_SOFT: '#A7A7A7',
  },
};

export const renderWithTheme = (component: any) => (
  <ThemeProvider theme={theme}>{component}</ThemeProvider>
);
