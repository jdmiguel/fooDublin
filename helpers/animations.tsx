import { keyframes, css } from 'styled-components';

const fadeIn = (endValue: number) => keyframes`
  0% { opacity: 0 }
  100% { opacity: ${endValue} }
`;

const fadeInFromPosY = (initialPosY: number) => keyframes`
  0% { opacity: 0; transform: translateY(${initialPosY}px); }
  100% { opacity: 1; ; transform: translateY(0); }
`;

const rotate = keyframes`
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
`;

const moveX = keyframes`
  0% { transform: translateX(-100%); }
  100% { transform: translateX(0); }
`;

export const fadeInAnimation = css<{ endValue?: number }>`
  animation: ${({ endValue }) => fadeIn(endValue || 1)} 0.15s linear;
`;

export const fadeInFromPosYAnimation = css<{
  initialPosY?: number;
  animationDuration?: number;
  animationDelay?: number;
}>`
  animation-name: ${({ initialPosY }) => fadeInFromPosY(initialPosY || 50)};
  animation-duration: ${({ animationDuration }) => animationDuration || 0.15}s;
  animation-delay: ${({ animationDelay }) => animationDelay || 0.25}s;
  animation-fill-mode: forwards;
  animation-timing-function: ease-out;
`;

export const circleLoaderAnimation = css`
  animation: ${rotate} 0.8s linear infinite;
`;

export const lineLoaderAnimation = css`
  animation: ${moveX} 1s ease-in-out infinite forwards;
`;
