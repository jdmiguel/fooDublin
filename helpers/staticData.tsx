import { CDN_URL_STATIC_DIRECTORY } from './utils';

export const DUBLIN_ID = 91;
export const BASE_URL = 'https://developers.zomato.com/api/v2.1/';

export const DEFAULT_TEXT_LOADING = 'Coming right up...';

export const MAX_RESTAURANT_DISPLAYED = 20;
export const MAX_RESTAURANT_RETRIEVED = 100;

export const MAX_SMALL_DEVICE_WIDTH = 640;

export const SCROLL_FACTOR = 4;
export const SCROLL_INITIAL_MOBILE_FACTOR = 10;
export const SCROLL_OFFSET_MOBILE_FACTOR = 4;
export const SCROLL_OFFSET_DESKTOP_FACTOR = 1.2;
export const SCROLL_DELAY = 4;

export const SHOWING_SCROLLUP_BUTTON_HEIGHT = 900;

export enum PlaceholderText {
  BLURRED = 'Search for locals...',
  FOCUSED = 'Start typing to search...',
}

export const DEFAULT_SUGGESTIONS = [
  {
    id: 0,
    imgSrc: '',
    firstText: '',
    secondText: '',
  },
];

export const LOCATIONS = [
  {
    id: 91,
    name: 'Dublin',
    path: 'dublin',
  },
  {
    id: 162239,
    name: 'South City West',
    path: 'south-city-west',
  },
  {
    id: 96201,
    name: 'Temple Bar',
    path: 'temple-bar',
  },
  {
    id: 95051,
    name: 'Clondalkin',
    path: 'clondalkin',
  },
  {
    id: 96181,
    name: 'Swords',
    path: 'swords',
  },
  {
    id: 96191,
    name: 'Tallaght',
    path: 'tallaght',
  },
  {
    id: 94881,
    name: 'Blanchardstown',
    path: 'blanchardstown',
  },
  {
    id: 162240,
    name: 'South City East',
    path: 'south-city-east',
  },
  {
    id: 95311,
    name: 'Dundrum',
    path: 'dundrum',
  },
  {
    id: 95341,
    name: 'Finglas',
    path: 'finglas',
  },
  {
    id: 94791,
    name: 'Ballyfermot',
    path: 'ballyfermot',
  },
  {
    id: 96081,
    name: 'Santry',
    path: 'santry',
  },
  {
    id: 95711,
    name: 'Lucan',
    path: 'lucan',
  },
  {
    id: 94671,
    name: 'Artane',
    path: 'artane',
  },
  {
    id: 95721,
    name: 'Malahide',
    path: 'malahide',
  },
  {
    id: 95301,
    name: 'Dun Laoghaire',
    path: 'dun-laoghaire',
  },
  {
    id: 96271,
    name: 'Walkinstown',
    path: 'walkinstown',
  },
  {
    id: 94871,
    name: 'Blackrock',
    path: 'blackrock',
  },
  {
    id: 95851,
    name: 'Palmerstown',
    path: 'palmerstown',
  },
  {
    id: 95981,
    name: 'Rathmines',
    path: 'rathmines',
  },
  {
    id: 94701,
    name: 'Balbriggan',
    path: 'balbriggan',
  },
  {
    id: 94761,
    name: 'Ballybrack',
    path: 'ballybrack',
  },
  {
    id: 96111,
    name: 'Smithfield',
    path: 'smithfield',
  },
  {
    id: 95181,
    name: 'Crumlin',
    path: 'crumlin',
  },
  {
    id: 94971,
    name: 'Castleknock',
    path: 'castleknock',
  },
  {
    id: 95511,
    name: 'Howth',
    path: 'howth',
  },
  {
    id: 94951,
    name: 'Cabra',
    path: 'cabra',
  },
  {
    id: 96291,
    name: 'Whitehall',
    path: 'whitehall',
  },
  {
    id: 95901,
    name: 'Portmarnock',
    path: 'portmarnock',
  },
  {
    id: 95641,
    name: 'Kimmage',
    path: 'kimmage',
  },
];

export const CUISINES = [
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/stew.svg`,
    id: 135,
    name: 'Irish',
    path: 'irish',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/pizza.svg`,
    id: 55,
    name: 'Italian',
    path: 'italian',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/bruschetta.svg`,
    id: 70,
    name: 'Mediterranean',
    path: 'mediterranean',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/rice.svg`,
    id: 3,
    name: 'Asian',
    path: 'asian',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/healthy_food.svg`,
    id: 143,
    name: 'Healthy Food',
    path: 'healthy-food',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/gulab.svg`,
    id: 148,
    name: 'Indian',
    path: 'indian',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/taco.svg`,
    id: 73,
    name: 'Mexican',
    path: 'mexican',
  },
  {
    iconSrc: `${CDN_URL_STATIC_DIRECTORY}/images/fries.svg`,
    id: 40,
    name: 'Fast Food',
    path: 'fast-food',
  },
];

export const HIGHLIGHTED_RESTAURANTS = [
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/cleaver-east.jpg`,
    id: 9100233,
    path: 'cleaver-east',
    name: 'Cleaver East Restaurant',
    description:
      'Situated in the heart of Ireland’s capital city, on East Essex Street in the popular Temple Bar area, Cleaver East Restaurant offers the finest Irish and European cuisine.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/tapas-de-lola.jpg`,
    id: 16518539,
    path: 'tapas-de-lola',
    name: 'Tapas de Lola',
    description:
      'This restaurant is a love letter to all things Spanish and flavorful. Its owners make regular pilgrimages to Spain for inspiration and pack it all into a lively menu of small plates.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/ely-wine.jpg`,
    id: 16517305,
    path: 'ely-wine',
    name: 'Ely Wine Bar',
    description:
      'Eric and his wife Michelle are the founders of this clubby place where you can eat and drink well both downstairs in the cozy basement and upstairs in an elegant dining room.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/oxmantown.jpg`,
    id: 16520229,
    path: 'oxmantown',
    name: 'Oxmantown',
    description:
      'In this small place, the art of a brilliant sandwich means every element is taken seriously, like the breakfast sandwich that comes with butcher Jack McCarthy’s black pudding.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/chapter-one.jpg`,
    id: 9100702,
    path: 'chapter-one',
    name: 'Chapter One',
    description:
      'Dublin writers are more closely associated with booze rather than food, but Chapter One is a Michelin-starred restaurant in the basement of the Dublin Writers Museum.',
  },
  {
    imgSrc: `${CDN_URL_STATIC_DIRECTORY}/images/rosa-madre.jpg`,
    id: 9101166,
    path: 'rosa-madre',
    name: 'Rosa Madre',
    description:
      'In the menu-hawking clamor of Temple Bar you might long for a quieter restaurant where you won’t feel like a tourist on a conveyor belt. If so, Rosa Madre is your place.',
  },
];

export const HIGHLIGHT_GENERIC_SRC = `${CDN_URL_STATIC_DIRECTORY}/images/highlight-generic.png`;
export const THUMB_GENERIC_SRC = `${CDN_URL_STATIC_DIRECTORY}/images/generic-thumb.png`;
export const DETAIL_GENERIC_SRC = `${CDN_URL_STATIC_DIRECTORY}/images/generic-detail.png`;
export const DEFAULT_RATING_STAR_LIST = [
  { id: 1, type: 'star_outline' },
  { id: 2, type: 'star_outline' },
  { id: 3, type: 'star_outline' },
  { id: 4, type: 'star_outline' },
  { id: 5, type: 'star_outline' },
];

export enum FilterSort {
  COST = 'cost',
  RANK = 'rating',
}
export enum FilterOrder {
  ASC = 'asc',
  DESC = 'desc',
}

export const FILTER_DATA = [
  {
    primaryText: 'cost',
    secondaryText: '- high to low',
    icon: 'keyboard_arrow_up',
    sort: FilterSort.COST,
    order: FilterOrder.ASC,
    id: 1,
  },
  {
    primaryText: 'cost',
    secondaryText: '- low to high',
    icon: 'keyboard_arrow_down',
    sort: FilterSort.COST,
    order: FilterOrder.DESC,
    id: 2,
  },
  {
    primaryText: 'rank',
    secondaryText: '- high to low',
    icon: 'keyboard_arrow_up',
    sort: FilterSort.RANK,
    order: FilterOrder.ASC,
    id: 3,
  },
  {
    primaryText: 'rank',
    secondaryText: '- low to high',
    icon: 'keyboard_arrow_down',
    sort: FilterSort.RANK,
    order: FilterOrder.DESC,
    id: 4,
  },
];
