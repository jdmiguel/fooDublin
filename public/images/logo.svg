<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 viewBox="-52 35.5 193.1 31.5" style="enable-background:new -52 35.5 193.1 31.5;" xml:space="preserve">
<style type="text/css">
	.st0{fill:#E5EAED;}
	.st1{fill:#EC7E00;}
</style>
<path class="st0" d="M-45,47.3h7.1c0.2,0.3,0.3,0.7,0.5,1.2s0.2,1,0.2,1.6c0,1-0.2,1.7-0.7,2.2c-0.4,0.4-1,0.7-1.8,0.7h-5v13.7
	c-0.3,0.1-0.7,0.1-1.3,0.2s-1.2,0.1-1.9,0.1c-1.4,0-2.5-0.2-3.1-0.7s-1-1.4-1-2.7V45.8c0-1.5,0.3-2.7,0.8-3.8s1.2-1.9,2.1-2.6
	s1.9-1.2,3.1-1.5s2.4-0.5,3.7-0.5c2.1,0,3.7,0.3,4.7,0.8c1,0.6,1.5,1.5,1.5,2.9c0,0.6-0.1,1.1-0.4,1.6c-0.2,0.5-0.5,0.8-0.8,1
	c-0.5-0.1-1.1-0.3-1.7-0.4s-1.3-0.2-2.1-0.2c-1.2,0-2.2,0.2-2.9,0.6s-1,1.1-1,2.2C-45,45.8-45,47.3-45,47.3z"/>
<path class="st0" d="M41.4,53c0,2.3-0.3,4.3-1,6.1c-0.7,1.7-1.7,3.2-3,4.3c-1.3,1.2-2.8,2-4.6,2.6c-1.8,0.6-3.8,0.9-6,0.9
	c-0.5,0-1.1,0-1.7,0s-1.3-0.1-2-0.2s-1.4-0.2-2-0.3c-0.7-0.1-1.3-0.3-1.9-0.5c-1.6-0.6-2.4-1.7-2.4-3.2V42.5c0-0.6,0.2-1.1,0.5-1.4
	s0.8-0.6,1.4-0.8c1.2-0.4,2.6-0.7,3.9-0.8c1.4-0.1,2.6-0.2,3.7-0.2c2.3,0,4.3,0.3,6.1,0.8c1.8,0.5,3.4,1.4,4.7,2.5s2.3,2.6,3.1,4.3
	C41,48.6,41.4,50.8,41.4,53z M33.6,53.3c0-2.6-0.6-4.6-1.8-5.9c-1.2-1.3-2.8-2-4.8-2c-0.4,0-0.8,0-1.2,0c-0.4,0-0.8,0.1-1.3,0.1
	v15.1c0.3,0.1,0.7,0.1,1.2,0.2c0.5,0,0.9,0.1,1.3,0.1c2,0,3.6-0.6,4.8-1.9C33,57.6,33.6,55.8,33.6,53.3z"/>
<path class="st0" d="M45.1,45.4c0.3-0.1,0.8-0.2,1.4-0.3c0.6-0.1,1.2-0.1,1.9-0.1c1.4,0,2.5,0.2,3.1,0.7s1,1.4,1,2.7V58
	c0,1.2,0.3,2,0.9,2.5s1.4,0.8,2.5,0.8c0.7,0,1.2-0.1,1.6-0.2c0.4-0.1,0.7-0.2,1-0.3V45.4c0.3-0.1,0.8-0.2,1.4-0.3
	c0.6-0.1,1.2-0.1,1.9-0.1c1.4,0,2.5,0.2,3.1,0.7s1,1.4,1,2.7v13.4c0,1.4-0.6,2.5-1.8,3.2c-1,0.6-2.2,1.1-3.6,1.5
	c-1.4,0.3-2.9,0.5-4.6,0.5c-1.6,0-3-0.2-4.3-0.5s-2.4-0.9-3.4-1.6s-1.7-1.7-2.2-2.8s-0.8-2.5-0.8-4.1L45.1,45.4L45.1,45.4z"/>
<path class="st0" d="M82,44.5c1.4,0,2.7,0.2,4,0.7s2.3,1.2,3.3,2.1c0.9,0.9,1.7,2.1,2.2,3.5s0.8,3,0.8,4.8S92,59,91.5,60.4
	c-0.6,1.4-1.3,2.6-2.4,3.5c-1,1-2.2,1.7-3.7,2.2s-3.1,0.8-4.8,0.8c-1.9,0-3.5-0.2-4.8-0.7c-1.3-0.4-2.5-1-3.4-1.7
	c-1.3-0.9-1.9-2-1.9-3.5V38c0.3-0.1,0.8-0.2,1.4-0.3s1.2-0.1,1.9-0.1c1.4,0,2.5,0.2,3.1,0.7s1,1.4,1,2.7v4.3
	c0.6-0.2,1.2-0.4,1.9-0.6C80.4,44.6,81.2,44.5,82,44.5z M80.6,50.4c-0.6,0-1.1,0.1-1.6,0.2s-0.9,0.3-1.3,0.6v9.4
	c0.3,0.2,0.7,0.3,1.1,0.5c0.4,0.1,1,0.2,1.6,0.2c1.3,0,2.3-0.5,3.1-1.4c0.8-0.9,1.2-2.3,1.2-4.1s-0.4-3.2-1.1-4
	C82.9,50.8,81.9,50.4,80.6,50.4z"/>
<path class="st0" d="M100.2,67c-1.4,0-2.5-0.2-3.1-0.7s-1-1.4-1-2.7V38.5c0.3-0.1,0.8-0.2,1.4-0.3s1.2-0.1,1.9-0.1
	c1.4,0,2.5,0.2,3.1,0.7s1,1.4,1,2.7v25.1c-0.3,0.1-0.8,0.2-1.4,0.3S100.9,67,100.2,67z"/>
<path class="st0" d="M108.4,46c0.3-0.1,0.8-0.2,1.4-0.3s1.2-0.1,1.9-0.1c1.4,0,2.5,0.2,3.1,0.7s1,1.4,1,2.7v17.6
	c-0.3,0.1-0.8,0.2-1.4,0.3c-0.6,0.1-1.2,0.1-1.9,0.1c-1.4,0-2.5-0.2-3.1-0.7s-1-1.4-1-2.7V46z"/>
<path class="st0" d="M141.2,66.6c-0.3,0.1-0.8,0.2-1.4,0.3c-0.6,0.1-1.2,0.1-1.9,0.1c-1.4,0-2.5-0.2-3.1-0.7s-1-1.4-1-2.7V53.4
	c0-0.9-0.3-1.6-0.8-2s-1.2-0.6-2.1-0.6c-0.6,0-1.1,0.1-1.6,0.2s-1,0.3-1.4,0.6v15.1c-0.3,0.1-0.8,0.2-1.4,0.3
	c-0.6,0.1-1.2,0.1-1.9,0.1c-1.4,0-2.5-0.2-3.1-0.7s-1-1.4-1-2.7v-13c0-0.8,0.2-1.4,0.5-1.9s0.8-0.9,1.4-1.3c1-0.7,2.2-1.2,3.7-1.7
	c1.5-0.4,3.1-0.6,5-0.6c3.3,0,5.8,0.7,7.5,2.1c1.8,1.4,2.6,3.4,2.6,6C141.2,53.3,141.2,66.6,141.2,66.6z"/>
<path class="st0" d="M12.4,51.3c-0.5-1.4-1.3-2.5-2.3-3.5S7.9,46,6.5,45.5s-2.9-0.8-4.6-0.8s-3.2,0.3-4.6,0.8s-2.6,1.2-3.6,2.2
	c-1,0.9-1.8,2.1-2.3,3.5s-0.8,2.9-0.8,4.6s0.3,3.3,0.8,4.7S-7.3,63-6.3,64c1,1,2.2,1.7,3.6,2.2S0.2,67,2,67s3.3-0.3,4.7-0.8
	S9.3,65,10.3,64s1.7-2.2,2.3-3.5c0.5-1.4,0.8-3,0.8-4.7S13.1,52.5,12.4,51.3z M2,64c-3.5,0-6.4-2.6-6.8-6H8.8C8.4,61.4,5.5,64,2,64z
	"/>
<path class="st0" d="M-13.1,51.3c-0.5-1.4-1.3-2.5-2.3-3.5c-1-1-2.2-1.8-3.6-2.3s-2.9-0.8-4.6-0.8s-3.2,0.3-4.6,0.8
	s-2.6,1.2-3.6,2.2c-1,0.9-1.8,2.1-2.3,3.5s-0.8,2.9-0.8,4.6s0.3,3.3,0.8,4.7s1.3,2.5,2.3,3.5c1,1,2.2,1.7,3.6,2.2s2.9,0.8,4.7,0.8
	s3.3-0.3,4.7-0.8s2.6-1.2,3.6-2.2c1-0.9,1.8-2.1,2.3-3.5s0.8-3,0.8-4.7S-12.4,52.5-13.1,51.3z"/>
<path class="st1" d="M-16.8,55.8c0,1-0.1,1.9-0.5,2.8c-0.2,0.9-0.7,1.6-1.3,2.1c-0.6,0.6-1.3,1-2.1,1.3s-1.7,0.5-2.8,0.5
	c-1.1,0-1.9-0.1-2.8-0.5c-0.9-0.2-1.6-0.7-2.1-1.3c-0.6-0.6-1.1-1.2-1.3-2.1c-0.2-0.9-0.5-1.8-0.5-2.8s0.1-1.9,0.5-2.8
	c0.2-0.9,0.7-1.6,1.3-2.1c0.6-0.6,1.3-1,2.1-1.3c0.9-0.2,1.7-0.5,2.7-0.5c1,0,1.8,0.1,2.7,0.5c0.9,0.4,1.6,0.7,2.1,1.3
	c0.5,0.6,1.1,1.2,1.3,2.1C-17,53.9-16.8,54.6-16.8,55.8z"/>
<path class="st0" d="M107.9,39.3c0-1.1,0.4-2,1.1-2.8s1.7-1.1,2.9-1.1s2.2,0.4,2.9,1.1c0.7,0.8,1.1,1.7,1.1,2.8s-0.4,2-1.1,2.8
	s-1.7,1.1-2.9,1.1s-2.2-0.4-2.9-1.1C108.3,41.3,107.9,40.4,107.9,39.3z"/>
</svg>
