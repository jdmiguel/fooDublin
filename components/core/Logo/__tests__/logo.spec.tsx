import React from 'react';
import { render } from '@testing-library/react';

import { Logo, LogoSize } from '../Logo';

import { LOGO_SRC_MOCK } from '../__mocks__/logo.mocks';

import { renderWithTheme } from '../../../../helpers/Theme';

describe('Component: Logo', () => {
  it('should render as big version', () => {
    const { container } = render(
      renderWithTheme(<Logo logoSrc={LOGO_SRC_MOCK} size={LogoSize.BIG} />),
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render as small version', () => {
    const { container } = render(
      renderWithTheme(<Logo logoSrc={LOGO_SRC_MOCK} size={LogoSize.SMALL} />),
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
