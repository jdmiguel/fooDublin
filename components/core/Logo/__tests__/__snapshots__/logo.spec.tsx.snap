// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Logo should render as big version 1`] = `
.c0 {
  width: 225px;
  margin: 15px 0 25px;
}

.c1 {
  max-width: 300px;
}

@media only screen and (min-width:768px) {
  .c0 {
    width: 270px;
  }
}

@media only screen and (min-width:992px) {
  .c0 {
    width: 300px;
    margin: 0 0 30px;
  }
}

<h1
  class="c0"
  data-testid="logo"
>
  <img
    alt="FooDublin Logo"
    class="c1"
    src="/public/images/logo.svg"
  />
</h1>
`;

exports[`Component: Logo should render as small version 1`] = `
.c0 {
  width: 190px;
}

.c1 {
  max-width: 300px;
}

@media only screen and (min-width:768px) {

}

@media only screen and (min-width:992px) {
  .c0 {
    width: 200px;
  }
}

<h1
  class="c0"
  data-testid="logo"
>
  <img
    alt="FooDublin Logo"
    class="c1"
    src="/public/images/logo.svg"
  />
</h1>
`;
